kolibri/plugins/user_profile/assets/src/views/ChangeFacility/MergeAccountDialog/index.vue

```diff
@@ -0,0 +1,243 @@
+<template>
+
+  <div>
+    <h1>{{ $tr('documentTitle') }}</h1>
+    <h2 data-test="fullName">
+      {{ fullName }}
+    </h2>
+    <h3 data-test="username">
+      {{ username }}
+    </h3>
+    <span v-if="usingAdminPasswordState">
+      <p>{{ mergeAccountUsingAdminAccount }}</p>
+
+      <KTextbox
+        v-model="formData.username"
+        data-test="usernameTextbox"
+        :autofocus="true"
+        :label="coreString('usernameLabel')"
+      />
+
+
+    </span>
+    <p v-else>
+      {{ mergeAccountUserInfo }}
+    </p>
+    <KTextbox
+      v-if="showPasswordTextbox"
+      ref="passwordTextbox"
+      v-model="formData.password"
+      data-test="passwordTextbox"
+      type="password"
+      :label="coreString('passwordLabel')"
+      :autofocus="true"
+      :invalid="isPasswordInvalid"
+      :invalidText="$tr('incorrectPasswordError')"
+      :floatingLabel="false"
+      @keydown.enter="handleContinue"
+    />
+    <div v-if="!usingAdminPasswordState">
+      {{ $tr('doNotKnowPassword') }}
+      <KButton
+        data-test="useAdminAccount"
+        :text="$tr('useAdminAccount')"
+        appearance="basic-link"
+        @click="useAdminAccount"
+      />
+    </div>
+    <BottomAppBar>
+      <slot name="buttons">
+        <KButtonGroup>
+          <KButton
+            :primary="false"
+            :text="coreString('backAction')"
+            appearance="flat-button"
+            data-test="backButton"
+            @click="sendBack"
+          />
+          <KButton
+            :primary="true"
+            :text="coreString('continueAction')"
+            data-test="continueButton"
+            @click="handleContinue"
+          />
+        </KButtonGroup>
+      </slot>
+    </BottomAppBar>
+
+  </div>
+
+</template>
+
+
+<script>
+
+  import commonCoreStrings from 'kolibri.coreVue.mixins.commonCoreStrings';
+  import BottomAppBar from 'kolibri.coreVue.components.BottomAppBar';
+  import { computed, inject, ref, watch } from 'kolibri.lib.vueCompositionApi';
+  import get from 'lodash/get';
+  import remoteFacilityUserData from '../../../composables/useRemoteFacility';
+
+  export default {
+    name: 'MergeAccountDialog',
+    metaInfo() {
+      return {
+        title: this.$tr('documentTitle'),
+      };
+    },
+    components: { BottomAppBar },
+
+    mixins: [commonCoreStrings],
+    setup(_, context) {
+      const changeFacilityService = inject('changeFacilityService');
+      const state = inject('state');
+
+      const isFormSubmitted = ref(false);
+      const isPasswordInvalid = ref(false);
+      const usingAdminPasswordState = ref(false);
+      // hack to get the $store in vue tests with composition api:
+      const store =
```

Comment:
yes, but didn't work inside the `setup` function of the component because `setup` is executed before anything else, including creating the component and assigning the store. I tested it a lot and was not able to find a better solution to make it work both in the app and when running the tests